lib/jose-jws-verify.js

```diff
@@ -0,0 +1,182 @@
+/*-
+ * Copyright 2015 Peculiar Ventures
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Handles signature verification.
+ *
+ * @param cryptographer  an instance of WebCryptographer (or equivalent). Keep
+ *                       in mind that decryption mutates the cryptographer.
+ * @param message        a JWS message
+ *
+ * @author Patrizio Bruno <patrizio@desertconsulting.net>
+ */
+JoseJWS.Verifier = function (cryptographer, message) {
+  "use strict";
+
+  var that = this,
+    alg,
+    jwt,
+    aad,
+    header,
+    payload,
+    signatures,
+    protectedHeader,
+    jwtRx = /^([0-9a-z_\-]+)\.([0-9a-z_\-]+)\.([0-9a-z_\-]+)$/i;
```

Comment:
The regexp still matches '\'. See https://github.com/PeculiarVentures/js-jose/commit/1d5ece09b7177c05cf70508db5cb03c008c588de#commitcomment-12703384